[DOC] ML-Commons agent framework updates (#10682)
opensearch-project/documentation-website: _ml-commons-plugin/agents-tools/agents/plan-execute-reflect.md

The agent currently supports re-evaluation only after each step. This allows the agent to dynamically adapt the plan based on intermediate results before proceeding to the next step.

> 

Reviewer: ```suggestion
```

The agent currently supports re-evaluation only after each step. This allows the agent to dynamically adapt the plan based on intermediate results before proceeding to the next step.

> For updates on the progress of the feature or if you want to leave feedback, see the associated [GitHub issue](https://github.com/opensearch-project/ml-commons/issues/3745).

Reviewer: ```suggestion
```

Example 1 - When you need to execute steps:
{
	"steps": ["This is an example step", "this is another example step"],
	"result": ""
}

Example 2 - When you have the final result:
{
	"steps": [],
	"result": "This is an example result\n with escaped special characters"
}
Important rules for the response:
1. Do not use commas within individual steps 
2. Do not add any content before or after the JSON 
3. Only respond with a pure JSON object 

When you deliver your final result, include a comprehensive report. This report must:
1. List every analysis or step you performed.
2. Summarize the inputs, methods, tools, and data used at each step.
3. Include key findings from all intermediate steps — do NOT omit them.
4. Clearly explain how the steps led to your final conclusion. Only mention the completed steps.
5. Return the full analysis and conclusion in the 'result' field, even if some of this was mentioned earlier. Ensure that special characters are escaped in the 'result' field.
6. The final response should be fully self-contained and detailed, allowing a user to understand the full investigation without needing to reference prior messages and steps.
```

> We recommend never modifying the response format instructions. If you intend to modify any prompts, you can inject the response format instructions by using `${parameters.plan_execute_reflect_response_format}` parameter.

Reviewer 2: ```suggestion
We do not recommend modifying the response format instructions. If you intend to modify any prompts, you can inject the response format instructions by using the `${parameters.plan_execute_reflect_response_format}` parameter.
```

### Planner template and prompt

To create a custom planner prompt template, modify the `planner_prompt_template` parameter.

**Introduced 3.0**
{: .label .label-purple }

The following template was used in version 3.0:

```json
${parameters.planner_prompt} \n Objective: ${parameters.user_prompt} \n ${parameters.plan_execute_reflect_response_format}
```

**Updated 3.2**
{: .label .label-purple }

The following template is used in version 3.2:

```json

> ${parameters.tools_prompt} \n ${parameters.planner_prompt} \n Objective: ${parameters.user_prompt} \n\n

Reviewer 3: is that a breaking change?
